File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/mapper/SysJobLogMapper.java
package com.vipamp.vipclaw.admin.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.vipamp.vipclaw.admin.entity.SysJobLog;
import org.apache.ibatis.annotations.Mapper;
import org.apache.ibatis.annotations.Param;
import org.apache.ibatis.annotations.Select;

import java.util.List;

/**
 * 定时任务日志 Mapper 接口
 *
 * @author vipamp
 * @since 2026-03-16
 */
@Mapper
public interface SysJobLogMapper extends BaseMapper<SysJobLog> {

    /**
     * 根据任务 ID 查询日志列表
     */
    @Select("SELECT * FROM sys_job_log WHERE job_id = #{jobId} ORDER BY create_time DESC")
    List<SysJobLog> selectByJobId(@Param("jobId") Long jobId);
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/mapper/SysJobLogMapper.kt
package com.vipamp.vipclaw.admin.mapper

import com.vipamp.vipclaw.admin.entity.SysJobLog
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param
import java.time.LocalDateTime

/**
 * SysJobLog Mapper interface
 */
@Mapper
interface SysJobLogMapper {

    // ==================== Basic CRUD Methods ====================

    fun selectById(@Param("id") id: Long): SysJobLog?

    fun insert(sysjoblog: SysJobLog): Int

    fun updateById(sysjoblog: SysJobLog): Int

    fun deleteById(@Param("id") id: Long): Int

    // ==================== Custom Query Methods ====================
    fun selectJobLogList(
        @Param("jobId") jobId: Long?,
        @Param("jobName") jobName: String?,
        @Param("status") status: Int?,
        @Param("startTime") startTime: LocalDateTime?,
        @Param("endTime") endTime: LocalDateTime?,
    ): List<SysJobLog>

    fun selectByJobId(@Param("jobId") jobId: Long): List<SysJobLog>
}
